plugin/pkg/admission/podpreset/exists/admission.go

```diff
@@ -0,0 +1,176 @@
+	// checked the cluster in the meantime. In theory this shouldn't matter,
+	// but if they succeeded and we got an error, we should just use their
+	// result.
+	l.Lock()
+	if l.podPresetsChecked {
+		l.Unlock()
+		return l.podPresetsExists, nil
+	}
+	// No errors, so we know  if the cluster supports PodPreset or not
+	if err == nil {
+		l.podPresetsExists = exists
+		l.podPresetsChecked = true
+	}
+	l.Unlock()
+	return exists, err
+}
+
+// checkClusterForPodPresets uses discovery to find if the cluster has the
+// right version of settings that supports PodPresets
+// This is a separate function from above since it does IO and we don't
+// want to hang on to a lock while doing this.
+func (l *exists) checkClusterForPodPresets() (bool, error) {
+	if resourceList, err := l.discovery.ServerResourcesForGroupVersion(SupportedPodPresetVersion); err != nil {
+		if errors.IsNotFound(err) {
+			glog.V(4).Infof("No PodPreset support in the cluster, not allowing PodPresetTemplates in Bindings")
+			// No such resource, which means the cluster does not support PodPresets
+			return false, nil
+		} else {
+			// Some other kind of error, so just return it
+			glog.V(4).Infof("ServerResourcesForGroupVersion failed: %s", err)
+			return false, err
+		}
+	} else {
+		for _, resource := range resourceList.APIResources {
+			if resource.Name == "podpresets" {
+				glog.V(4).Infof("PodPreset support found, allowing PodPresetTemplates in Bindings")
+				return true, nil
+			}
+		}
+	}
+	return false, nil
+}
+
+func (l *exists) Admit(a admission.Attributes) error {
+	// We only care about updates / creates
+	if a.GetOperation() != admission.Create && a.GetOperation() != admission.Update {
+		return nil
+	}
+
+	// We only care about bindings
+	if a.GetResource().Group != servicecatalog.GroupName || a.GetResource().GroupResource() != servicecatalog.Resource("bindings") {
+		return nil
+	}
+	binding, ok := a.GetObject().(*servicecatalog.Binding)
+	if !ok {
+		return errors.NewBadRequest("Resource was marked with kind Binding but was unable to be converted")
+	}
+	if binding.Spec.AlphaPodPresetTemplate == nil {
+		return nil
+	}
+
+	// Request is for creating a binding with pod preset, check to see if the cluster supports it
+	ppExists, err := l.hasSupportForPodPresets()
+	if err != nil {
+		return errors.NewInternalError(err)
+	}
+
+	if ppExists {
+		return nil
+	} else {
+		return admission.NewForbidden(a, fmt.Errorf("Unable to create a binding with PodPresetTemplate because the cluster does not support PodPreset. Need support for %s resource", SupportedPodPresetVersion))
+	}
+}
+
+// NewExists creates a new admission control handler that checks for existence of PodPresets
+// before allowing a Binding to be created with PodPresetTemplate.
+func NewExists() (admission.Interface, error) {
+	return &exists{
+		Handler: admission.NewHandler(admission.Create, admission.Update),
+	}, nil
+}
+
+func (l *exists) SetKubeClientSet(client kubeclientset.Interface) {
```

Comment:
i would prefer we follow the wait for ready pattern.

see: 
https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/admission/namespace/exists/admission.go#L98
https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/admission/namespace/exists/admission.go#L61

so for example, do not allow any requests to come in until you can determine the result by adding a ReadyFunc, and have that ReadyFunc do the discovery check once.